sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -371,16 +431,24 @@ class SessionCatalog(
    */
   def getTempViewOrPermanentTableMetadata(name: TableIdentifier): CatalogTable = synchronized {
```

Comment:
So `getTableMetadataOption` is only responsible for retrieving metadata of tables from an external catalog? The method names are a littble bit confusing.